packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -6,32 +6,100 @@ import { some, groupBy } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { CheckboxControl, Modal, Button } from '@wordpress/components';
+import {
+	CheckboxControl,
+	Button,
+	PanelBody,
+	PanelRow,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { useSelect, useDispatch } from '@wordpress/data';
-import { useState } from '@wordpress/element';
+import { useState, useCallback } from '@wordpress/element';
+import { close, page, layout, grid, blockDefault } from '@wordpress/icons';
+
+function EntityRecordState( { record, checked, onChange, closePanel } ) {
+	const { name, kind, title, key } = record;
+	const parentBlockId = useSelect( ( select ) => {
+		// Get entity's blocks.
+		const { blocks = [] } = select( 'core' ).getEditedEntityRecord(
+			kind,
+			name,
+			key
+		);
+		// Get parents of the entity's first block.
+		const parents = select( 'core/block-editor' ).getBlockParents(
+			blocks[ 0 ]?.clientId
```

Comment:
What happens if the entity has no blocks? E.g. an empty template part.

Reply:
I guess a couple things regarding that:

An empty template part doesn't seem to trigger as being dirty until it has a block.  It seems clicking on the new template part initiates the paragraph block by default, but at this point it is no longer empty.

If that empty template part WERE to be dirty (hypothetically), this would pass `undefined` into getBlockParents which would then return an empty array.  From there since we don't have the parentId, the 'Select Entity' button would not be present.  I suppose this behavior would make sense, as if the template part has no blocks selecting it would not really make sense.  🤔

Reply:
That makes sense. Ideally, we would select the wrapper. This could happen when someone completely clears a template part. It's a super edge case, so let's not worry about it for now.